modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/web/DetectorMappingController.java

```diff
@@ -53,8 +63,11 @@ public DetectorMapping getDetectorMapping(@RequestParam String id) {
 
     @RequestMapping(method = RequestMethod.POST)
     @ResponseStatus(HttpStatus.CREATED)
-    public String createDetectorMapping(@RequestBody CreateDetectorMappingRequest request) {
+    public String createDetectorMapping(@RequestBody CreateDetectorMappingRequest request, @RequestHeader HttpHeaders header) {
+        SpanContext parentSpanContext = trace.extractParentSpan(header);
+        Span span = trace.startSpan("create-mappings", parentSpanContext);
         request.validate();
+        span.finish();
```

Comment:
the span got finished before actually creating the mapping - so the duration of the span will not be accurate, this should be moved to after `createDetectorMapping` call in the next line (assign the return value of `detectorMappingRepo.createDetectorMapping` to a variable, and return that variable after finishing the span)

Reply:
Fixed.